src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -349,40 +355,40 @@ public AzureVMDeploymentInfo createDeployment(
                         if (msiEnabled) {
                             ObjectNode identityNode = mapper.createObjectNode();
                             identityNode.put("type", "systemAssigned");
-                            ObjectNode.class.cast(resource).replace("identity", identityNode);
+                            ((ObjectNode) resource).replace("identity", identityNode);
                         }
                         if (osDiskSizeChanged) {
                             JsonNode jsonNode = resource.get("properties").get("storageProfile").get("osDisk");
-                            ObjectNode.class.cast(jsonNode).replace("diskSizeGB", new IntNode(osDiskSize));
+                            ((ObjectNode) jsonNode).replace("diskSizeGB", new IntNode(osDiskSize));
                         }
                         if (availabilitySetEnabled) {
                             ObjectNode availabilitySetNode = mapper.createObjectNode();
                             availabilitySetNode.put("id", String.format(
                                     "[resourceId('Microsoft.Compute/availabilitySets', '%s')]", availabilitySet));
                             JsonNode propertiesNode = resource.get("properties");
-                            ObjectNode.class.cast(propertiesNode).replace("availabilitySet",
+                            ((ObjectNode) propertiesNode).replace("availabilitySet",
                                     availabilitySetNode);
                         }
                     }
                 }
             }
 
-            copyVariableIfNotBlank(tmp, properties, "imageId");
-            copyVariableIfNotBlank(tmp, properties, "imagePublisher");
-            copyVariableIfNotBlank(tmp, properties, "imageOffer");
-            copyVariableIfNotBlank(tmp, properties, "imageSku");
-            copyVariableIfNotBlank(tmp, properties, "imageVersion");
-            copyVariableIfNotBlank(tmp, properties, "osType");
-            putVariableIfNotBlank(tmp, "image", template.getImage());
+            putVariableIfNotBlank(tmp, "imageId", template.getImageReference().getId());
+            putVariableIfNotBlank(tmp, "imagePublisher", template.getImageReference().getPublisher());
```

Comment:
These names have been changed in the properties map.

Reply:
What do you mean? What are you suggesting?
This maps to the ARM template:
https://github.com/timja/azure-vm-agents-plugin/blob/aa265a30159dd3c226af3036d97f8f293285c0f0/src/main/resources/referenceImageTemplateWithScript.json#L91

Reply:
`properties` is map generated by `AzureVMAgentTemplate.getTemplateProperties(template)` method, you have changed the key name in the map, so there will be no `imageId`... keys.

Reply:
This line isn't using properties, 

I've undone the overzealous rename, but I can't find any code using properties for those properties.

![image](https://user-images.githubusercontent.com/21194782/57357084-096b2680-716a-11e9-9e06-8a2196a3af5e.png)

Reply:
This should use properties values for `AzureVMAgentTemplate.getTemplateProperties(template)` will add some default values which will be used for basic image configuration. Method like `template.getImageReference().getOffer()` will get null or empty string.

Reply:
I think I've pushed what you meant